pkg/apis/core/validation/conditional_validation.go

```diff
@@ -226,3 +228,29 @@ func podSCTPFields(podSpec *api.PodSpec, fldPath *field.Path) []*field.Path {
 	}
 	return fields
 }
+
+// validateMixedProtocolLBSVC checks if the old Service has type=LoadBalancer and whether the Service has different Protocols
+// on its ports. If the MixedProtocolLBSVC feature flag is disabled the usage of different Protocols in the new Service is
+// valid only if the old Service has different Protocols, too.
+func validateMixedProtocolLBSVC(service, oldService *api.Service) (errs field.ErrorList) {
+	if service.Spec.Type != api.ServiceTypeLoadBalancer {
+		return
+	}
+	if !utilfeature.DefaultFeatureGate.Enabled(features.MixedProtocolLBSVC) {
+		if serviceHasMixedProtocols(service) && !serviceHasMixedProtocols(oldService) {
+			errs = append(errs, field.Invalid(field.NewPath("spec.ports"), service.Spec.Ports, "cannot create an external load balancer with mix protocols"))
```

Comment:
nit: "may not contain more than 1 protocol when `type` is 'LoadBalancer'"

Reply:
note the quoting

Reply:
Fixed.